src/PackageManagement.VisualStudio/Utility/EnvDTEProjectUtility.cs

```diff
@@ -625,7 +640,9 @@ public static string GetTargetFrameworkString(EnvDTEProject envDTEProject)
                 ||
                 string.Equals(envDTEProject.Kind, NuGetVSConstants.FsharpProjectTypeGuid, StringComparison.OrdinalIgnoreCase)
                 ||
-                string.Equals(envDTEProject.Kind, NuGetVSConstants.JsProjectTypeGuid, StringComparison.OrdinalIgnoreCase))
+                string.Equals(envDTEProject.Kind, NuGetVSConstants.JsProjectTypeGuid, StringComparison.OrdinalIgnoreCase)
+                ||
+                string.Equals(envDTEProject.Kind, NuGetVSConstants.CpsProjectTypeGuid, StringComparison.OrdinalIgnoreCase))
```

Comment:
OK. I've determined that CPS is (correctly) returning S_OK and pFound==1 from IsDocumentInProject in this case. However evidently NuGet interprets that as meaning that it doesn't need to add the file to the project, which is incorrect. Your return statement on the last line of this method only checks `Succeeded(hr) && pFound == 1`. But it should also check that `priority[0] >= VSDOCUMENTPRIORITY.DP_STANDARD`. See, in this case you've already created the file, so yes CPS can find it, but it's passing back `DP_NONMEMBER` which means it's not included in the project and you need to add it, but you pass in an empty array instead of a one-element array and considering the result.
Can you fix that and revert your change to this `if` check that branches into DTE?